### PR TITLE
[tests-only][full-ci]added test to create text file with `/app/new` endpoint

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
Added test to create text file with `/app/new` endpoint.

## Related Issue

- https://github.com/owncloud/ocis/issues/9682

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 1]
```diff
@@ -432,3 +432,25 @@ Feature: collaboration (wopi)
       | app-endpoint                                              | url-query       |
       | /app/open-with-web?file_id=<<FILEID>>&app_name=FakeOffice | app=FakeOffice& |
       | /app/open-with-web?file_id=<<FILEID>>                     |                 |
+
+
+  Scenario: create text file using app endpoint
+    Given user "Alice" has created folder "testFolder"
+    When user "Alice" creates file "testfile.txt" inside folder "testFolder" in space "Personal" using app endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:439]: should we say?
```suggestion
    When user "Alice" creates file "testfile.txt" inside folder "testFolder" in space "Personal" using wopi endpoint
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 2]
```diff
@@ -432,3 +432,25 @@ Feature: collaboration (wopi)
       | app-endpoint                                              | url-query       |
       | /app/open-with-web?file_id=<<FILEID>>&app_name=FakeOffice | app=FakeOffice& |
       | /app/open-with-web?file_id=<<FILEID>>                     |                 |
+
+
+  Scenario: create text file using app endpoint
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:437]: How add adding tests for other drive space/share if possible.

Comment (PrajwolAmatya) [reply tests/acceptance/features/apiCollaboration/wopi.feature:437]: I think for text file that would not be necessary.

Comment (amrita-shrestha) [reply tests/acceptance/features/apiCollaboration/wopi.feature]: in API tests, we cover every edge case. So, IMO covering every type of drive goes with our principle until there is limitation.

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 3]
```diff
@@ -493,3 +493,189 @@ Feature: collaboration (wopi)
...
+            "type": "string",
+            "pattern": "^%file_id_pattern%$"
+          }
+        }
+      }
+      """
+    And as "Alice" file "testFolder/testfile.txt" should exist
+
+
+  Scenario Outline: sharee creates text file inside shared folder as Editor/Uploader
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:520]:
```suggestion
  Scenario Outline: sharee with permission Editor/Uploader creates text file inside shared folder
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 4]
```diff
@@ -493,3 +493,189 @@ Feature: collaboration (wopi)
...
+      """
+    And as "Alice" file "testFolder/testFile.txt" should exist
+    And as "Brian" file "Shares/testFolder/testFile.txt" should exist
+    Examples:
+      | permissions-role |
+      | Editor           |
+      | Uploader         |
+
+
+  Scenario: sharee creates text file inside shared folder as Viewer
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:553]:
```suggestion
  Scenario: sharee with Viewer tries to creates text file inside shared folder
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 5]
```diff
@@ -493,3 +493,189 @@ Feature: collaboration (wopi)
...
+    Given user "Alice" has created folder "testFolder"
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | testFolder |
+      | space           | Personal   |
+      | sharee          | Brian      |
+      | shareType       | user       |
+      | permissionsRole | Viewer     |
+    When user "Brian" creates file "testFile.txt" inside folder "testFolder" in space "Shares" using wopi endpoint
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:561]:
```suggestion
    When user "Brian" tries to creates file "testFile.txt" inside folder "testFolder" in space "Shares" using wopi endpoint
```
their should be tries to  step to test b=negative testing

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 6]
```diff
@@ -493,3 +493,189 @@ Feature: collaboration (wopi)
...
+            "const": "error calling InitiateFileUpload"
+          }
+        }
+      }
+      """
+    And as "Alice" file "testFolder/testFile.txt" should not exist
+    And as "Brian" file "Shares/testFolder/testFile.txt" should not exist
+
+
+  Scenario: create text file project space using wopi endpoint
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:585]:
```suggestion
  Scenario: create text file in project space using wopi endpoint
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 7]
```diff
@@ -493,3 +493,189 @@ Feature: collaboration (wopi)
...
+            "pattern": "^%file_id_pattern%$"
+          }
+        }
+      }
+      """
+    And for user "Alice" folder "testFolder" of the space "new-space" should contain these files:
+      | testFile.txt |
+
+
+  Scenario Outline: sharee creates text file inside shared space using wopi endpoint as Space Editor/Manager
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:611]:
```suggestion
  Scenario Outline: user with SpaceEditor/Manager role creates a text file inside a project space using wopi endpoint
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 8]
```diff
@@ -493,3 +493,189 @@ Feature: collaboration (wopi)
...
+      | testFile.txt |
+    And for user "Brian" folder "testFolder" of the space "new-space" should contain these files:
+      | testFile.txt |
+    Examples:
+      | permissions-role |
+      | Space Editor     |
+      | Manager          |
+
+
+  Scenario: sharee creates text file inside shared space using wopi endpoint as Viewer
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:648]:
```suggestion
Scenario Outline: user with Viewer role tries to create a text file inside a project space using wopi endpoint
```

**Review by amrita-shrestha — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 9]
```diff
@@ -522,3 +522,189 @@ Feature: collaboration (wopi)
...
+            "const": "error calling InitiateFileUpload"
+          }
+        }
+      }
+      """
+    And as "Alice" file "testFolder/testFile.txt" should not exist
+    And as "Brian" file "Shares/testFolder/testFile.txt" should not exist
+
+
+  Scenario: create text file in project space using wopi endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:614]: this should be covered by scenario below

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 10]
```diff
@@ -522,3 +522,189 @@ Feature: collaboration (wopi)
...
+            "pattern": "^%file_id_pattern%$"
+          }
+        }
+      }
+      """
+    And for user "Alice" folder "testFolder" of the space "new-space" should contain these files:
+      | testFile.txt |
+
+
+  Scenario Outline: user with SpaceEditor/Manager role creates a text file inside a project space using wopi endpoint
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiCollaboration/wopi.feature:640]:
```suggestion
  Scenario Outline: user with Space Editor/Manager role creates a text file inside a project space using wopi endpoint
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 11]
```diff
@@ -522,3 +522,189 @@ Feature: collaboration (wopi)
...
+    Given using spaces DAV path
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "new-space" with the default quota using the Graph API
+    And user "Alice" has created a folder "testFolder" in space "new-space"
+    And user "Alice" has sent the following space share invitation:
+      | space           | new-space    |
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | Space Viewer |
+    When user "Brian" tries to creates file "testFile.txt" inside folder "testFolder" in space "new-space" using wopi endpoint
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiCollaboration/wopi.feature:687]:
```suggestion
    When user "Brian" tries to create a file "testFile.txt" inside folder "testFolder" in space "new-space" using wopi endpoint
```

File: tests/acceptance/features/apiCollaboration/wopi.feature (hunk 9 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiCollaboration/wopi.feature:614]: the scenario below is for the shared project space, where user with different share role creating file inside the shared project space.

Comment (saw-jan) [reply tests/acceptance/features/apiCollaboration/wopi.feature:614]: then we might need to mention space admin in this scenario

**Review by saw-jan — APPROVED** [review]
LGTM

**Comment by prashant-gurung899** [comment]
LGTM :+1:

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 12]
```diff
@@ -522,3 +522,189 @@ Feature: collaboration (wopi)
...
+      """
+    And as "Alice" file "testFolder/testFile.txt" should exist
+    And as "Brian" file "Shares/testFolder/testFile.txt" should exist
+    Examples:
+      | permissions-role |
+      | Editor           |
+      | Uploader         |
+
+
+  Scenario: sharee with Viewer tries to create text file inside shared folder
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiCollaboration/wopi.feature:582]: What is viewer? plz clarify that, I see some inconsistencies in **role and permission** word uses. Please enlight me on that too

Comment (PrajwolAmatya) [reply tests/acceptance/features/apiCollaboration/wopi.feature]: just missed the wording.

**Review by amrita-shrestha — APPROVED** [review]

**Comment by anon-pradip** [comment]
LGTM :+1:
